src/ssl.c

```diff
@@ -15537,6 +15646,101 @@ char* wolfSSL_ASN1_TIME_to_string(WOLFSSL_ASN1_TIME* t, char* buf, int len)
 }
 #endif /* WOLFSSL_MYSQL_COMPATIBLE */
 
+#if defined(OPENSSL_EXTRA) && !defined(NO_ASN_TIME) \
+&& !defined(USER_TIME) && !defined(TIME_OVERRIDES)
+
+  #if !defined(XSNPRINTF)
+      #define XSNPRINTF snprintf
```

Comment:
I think this may warrant further investigation.  ssl.c includes <wolfssl/internal.h> which then includes <wolfssl/wolfcrypt/types.h>.  If STRING_USER is not defined, types.h should define XSPRINTF.  In the case where STRING_USER is defined, the user will be defining XSNPRINTF to their own function.